include/libcyphal/transport/transfer_id_generators.hpp

```diff
@@ -0,0 +1,141 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_TRANSFER_ID_GENERATORS_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_TRANSFER_ID_GENERATORS_HPP_INCLUDED
+
+#include "types.hpp"
+
+#include <cetl/cetl.hpp>
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <bitset>
+#include <cstddef>
+#include <utility>
+
+namespace libcyphal
+{
+namespace transport
+{
+
+/// Internal implementation details of the Presentation layer.
+/// Not supposed to be used directly by the users of the library.
+///
+namespace detail
+{
+
+/// @brief Defines a trivial transfer ID generator.
+///
+/// The generator is trivial in the sense that it simply modulo increments the transfer ID.
+/// B/c modulo is expected to be quite big (like >= 2^48), collisions of transfer ids are unlikely.
+/// Normally in use for UDP transport, where the modulo is `2^64 - 1`.
```

Comment:
Do we really need to bother with the modulo though? $2^{48}$ is already large enough to be unreachable for all practical scenarios, so we can just use plain increments.